### ISSUE TITLE
Additional point elements not in the VSA-DSS model

### ISSUE THREAD

**Issue by andreasneumann** [qgep/QGEP#103]
In Uster we have a number of point elements that should appear in the "Werkplan Abwasser" (map view) when there is a change between two neighboring sections. These could be automatically derived through a Trigger, but should be stored somewhere for performance reasons:

This includes:

* Change in construction year (could be derived automatically)
* change in slope - german Gefällsbruch (this often can't be derived automatically, since sections in the secondary systems don't carry elevation attributes)
* change in material (could be derived automatically)
* change in diameter (could be derived automatically)
* connection point to symbolize when two sections are connected without a manhole and when none of the above apply

See also attached image
![ausschnitt_werkplan_uster](https://cloud.githubusercontent.com/assets/884476/7495280/41c20c76-f40d-11e4-8620-a20e3bdf6f62.png)

**Comment by sjib** [comment]
The first four cases (change of ...) are always at a node (Abwasserknoten). You could either add an attribute to node with these four cases for labelling (yes/no) - some of these cases could be even at the same node. I would suggest to generate them at least in the primary system (PAA) with a trigger. It can still be changed manually, but then you can have wrong attributation, that missmatch the situation.
node.change_construction_year
node.change_slope
node.change_material
node.change_diameter

For the secondary system I would also add a trigger and then asks the question for assigning the case change_slope if not enogh data is available. 

The fith case (Abzweiger) is also identified with a Haltung (reach) - Haltung (reach) connection in the secondary system - then you would have to attribute the to node point of the first reach.
Rule: if saa then if reachpoint.fk_network_element = reach then set reachpoint.abzweiger = true

**Comment by andreasneumann** [comment]
I don't believe the above suggestions are only interesting for Uster. I do believe that most other organizations need them as well. SIA Geo405 Merkblatt 2015 mentions those on page 62. Their remark is that these point elements can be derived from the relations in the model. So I would strongly vote to include those into our project.

**Comment by m-kuhn** [comment]
Opened a separate issue #105 for my question which seems to be unrelated.

Question: Do these points only need to be shown in large scale situations? If yes they can probably be implemented by means of a view (Sorry, you know me :) ). We can still switch to triggers and calculated fields if it turns out to be a bad idea.

And some thoughts from Urs:

> Hallo zäme
> 
> Hier noch ein paar Gedanken auf deutsch :-)
> 
> Ich habe solche Punkte auch schon dargestellt.
> 
> Methode 1: ich lasse mir Abwasserknoten, welche als Abwasserbauwerk einen Kanal (also nicht ein Schachtbauwerk) haben, darstellen. Damit erhalte ich Gefälls-, Material-, Nennweiten-, Baujahrswechsel in den PAA-Kanälen. Allerdings kann ich die Art des Wechsels damit nicht einfach ermitteln. Und da an einer Stelle mehrere Wechsel möglich sind, wird es schon sehr komplex. Allerdings erlaube ich mir die Frage, ob wirklich jeder Wechsel ein eigenes Symbol haben muss. 
> 
> In den SAA-Leitungen mache ich aber bei den *-wecheln keine Abwasserknoten. Also kann ich das dort nicht einfach darstellen.
> 
> Methode 2: In Thun habe ich bei Blindanschlüssen mit einem Symbol (kleiner Punkt mit weissem Rand) dargestellt, dass die Lage des Blindanschlusses aus Kanalfernsehaufnahmen oder Einmassen vermessen (und damit lagerichtig) ist. Wo der Blindanschluss nur etwa digitalisiert (geschätzt) ist, da ist kein solcher Punkt. Die Unterschiede machte ich, in dem ich beim to-Haltungspunkt im Feld Bemerkung den Eintrag "Blindanschluss bei xx" machte. xx steht hier für die Distanz aus den Kanalfernsehaufnahmen. Wenn ich den Blindanschluss mit dem Mass xx auf dem Kanal ab dem Startschacht mit dem "Entlang-Tool"konstruiert habe, dann hat mit ein kleines Zusatzscript den Eintrag in die Dabase automatisch gemacht. Masse gegen die Fliessrichtung sind dabei negativ. Die Idee war dann, dass ich damit diesen Blindanschluss später auch automatisch wieder neu berechnen könnte (wenn z.Bsp. aus einer neueren Vermessung ein Schacht oder Kanal plötzlich eine korrigierte Lage haben sollte). Dazu muss ich das Entlang-Mass irgendwo speichern (im Datenmodell wurde das auch schon diskutiert, ist aber nicht vorgesehen).
> 
> Soweit eigentlich problemlos. Ich glaube kaum, dass das Bemerkungsfeld der Haltungspunkte für sonst etwas gebraucht wird.
> 
> Man könnte also bei den gewünschten Punkten (und Haltungspunkte hat es zwingend immer bei den *-wechseln oder beim verschlossenen Haltungsende) eine entsprechende Bemerkung machen und dann diejenigen Punkte mit einem Eintrag entsprechend dem Eintrag symbolisieren...
> 
> Geht aber nicht beim Fallstrang. Ich nehme an, damit meinen auch die Ustener (oder Usterer :-)) das Trockenwetterfallrohr ausserhalb dem Schacht (oder auch solche innerhalb dem Schacht). Das ist ja ein Bauwerksteil auf dem Kanal. Die Lage selbst auf dem Kanal ist an sich nicht definiert. Aber das wird immer unmittelbar vor dem Schacht sein. Man müsste also auf den Haltungen mit Bauwerksteil Trockenwetterfallrohr etwa 1.5m vor dem Haltungsende ein solches kleines Kreislein erzeugen. Ich nehme an, damit wären die meisten zufrieden und das ist keine Hexerei (das ist etwa dort, wo ich jeweilen die Beschriftung E1, E2 etc platziere, wenn es mehrere Einläufe hat und die Einlaufkoten beschriftet werden).
> 
> Noch zu den Symbolen: müssen es zwingend die gleichen Symbole sein, welche bereits in der GKP- und Schichtplanzeit benutzt wurden, als Pläne noch eine Lebensdauer von 20 Jahren hatten. Dann stellt ihr recht hohe Ansprüche an die Darstellung. Ansprüche, die z.Bsp. die zwiefellos sehr gute automatische Beschriftung von QGIS auch nicht erfüllen kann, wenn der Platz mal eng wird. Man kann dann ja nicht einfach eine Beschriftung weglassen. Aber das ist ein anderes Thema. Einfach noch zu den Symbolen: man kann sich das Problem der Orientierung auch ersparen, wenn man entsprechende (punktsymmetrische) Symbole wählt. Ein Kreis, Punkt, Stern... braucht keine Orientierung. Man muss sich etwas umgewöhnen, aber es geht.
> 
> e schöne Hinechd
> 
> Urs

**Comment by m-kuhn** [comment]
Add most of them to the wastewater nodes (see Word file)

**Comment by ponceta** [comment]
I've stored these in a customization of my wastewater nodes :

https://github.com/VilleDePully/datamodel/blob/c86d6fc4f45161575678dbd1ae14d488a4777d62/delta/pully_custom.sql#L25-L50

++
Easy to include, 
inheritance of hierarchy and usage

--
Can introduce nodes where the VSA does not want nodes (secondary reach junctions) (could be filtered at export)

**Comment by rbovard** [comment]
Thanks @ponceta!

I've the same needs here, I'll take your customization.
